src/facets/AccusationLibrary.sol

```diff
@@ -1,40 +1,76 @@
     }
-    
-    //
-    //
-    //
+
+    /// @notice This function validates an accusation of multiple proposals.
+    /// @param _signature0 The signature of pclaims0
+    /// @param _pClaims0 The PClaims of the accusation
+    /// @param _signature1 The signature of pclaims1
+    /// @param _pClaims1 The PClaims of the accusation
+    /// @return the address of the signer
     function AccuseMultipleProposal(
-        bytes calldata _signatureGroup0,
+        bytes calldata _signature0,
         bytes calldata _pClaims0,
-        bytes calldata _signatureGroup1,
+        bytes calldata _signature1,
         bytes calldata _pClaims1
-    ) internal {
-        
+    ) internal view returns(address) {
+        // ecrecover sig0/1 and ensure both are valid and accounts are equal
+        address signerAccount0 = recoverMadNetSigner(_signature0, _pClaims0);
+        address signerAccount1 = recoverMadNetSigner(_signature1, _pClaims1);
+
+        require(signerAccount0 == signerAccount1, "Invalid multiple proposal accusation, the signers of the proposals are different!");
+
+        // ensure the hashes of blob0/1 are different
+        require(keccak256(_pClaims0) != keccak256(_pClaims1), "Invalid multiple proposal accusation, the PClaims are equal!");
+
+        PClaimsParserLibrary.PClaims memory pClaims0 = PClaimsParserLibrary.extractPClaims(_pClaims0, PClaimsParserLibrary.CAPNPROTO_HEADER_SIZE);
+        PClaimsParserLibrary.PClaims memory pClaims1 = PClaimsParserLibrary.extractPClaims(_pClaims1, PClaimsParserLibrary.CAPNPROTO_HEADER_SIZE);
+
+        // ensure the height of blob0/1 are equal using RCert sub object of PClaims
+        require(pClaims0.rCert.rClaims.height == pClaims1.rCert.rClaims.height, "Invalid multiple proposal accusation, the block heights between the proposals are different!");
+
+        // ensure the round of blob0/1 are equal using RCert sub object of PClaims
+        require(pClaims0.rCert.rClaims.round == pClaims1.rCert.rClaims.round, "Invalid multiple proposal accusation, the round between the proposals are different!");
+
+        // ensure the chainid of blob0/1 are equal using RCert sub object of PClaims
+        require(pClaims0.rCert.rClaims.chainId == pClaims1.rCert.rClaims.chainId, "Invalid multiple proposal accusation, the chainId between the proposals are different!");
+
+        // ensure the chainid of blob0 is correct for this chain using RCert sub object of PClaims
+        uint32 chainId = ParticipantsLibrary.participantsStorage().chainId;
+        require(pClaims0.rCert.rClaims.chainId == chainId, "Invalid multiple proposal accusation, the chainId is invalid for this chain!");
+
+        // ensure both accounts are applicable to a currently locked validator - Note<may be done in different layer?>
+        require(ParticipantsLibrary.isValidator(signerAccount0), "Invalid multiple proposal accusation, the signer of these proposals is not a valid validator!");
+
+        return signerAccount0;
```

Comment:
Why do we return an address?

Reply:
If I understood correctly, we need to know the address to the bad validator (who signed the multiple proposals) in order to punish him. The one doing the accusation transaction can be (and probably will be) a different person. Did I get the idea wrong?